### PR TITLE
Add ItemDisplayProperties#addLore(int, String...) and #addLore(String…)

### PR DESCRIPTION
**Pull request by nickrobson**
This is a feature that should be there from the start, as it simplifies the development process by a lot.

It's also something that should have been in Bukkit and I wish it was.

### CODE REVIEW

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 1]
```diff
@@ -56,4 +56,30 @@
     default void setLore(String... lore) {
         setLore(Arrays.asList(lore));
     }
+    
+    /**
+     * Adds lore to the item starting from a specific index.
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: White space needed

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 2]
```diff
@@ -56,4 +56,30 @@
...
+     * @param index Index to begin at.
+     * @param lore Lore to be added.
+     * @see #addLore(String...)
+     */
+    default void addLore(int index, String... lore) {
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: Cleanup indentation

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 3]
```diff
@@ -56,4 +56,30 @@
...
+    	setLore(curr);
+    }
+    
+    /**
+     * Appends lore to the item.
+     * @param lore Lore to be appended.
+     * @see #addLore(int, String...)
+     */
+    default void addLore(String... lore) {
+    	addLore(lore().size(), lore);
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: Cleanup indentation

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 4]
```diff
@@ -56,4 +56,30 @@
...
+    	if (curr.isEmpty() || index == curr.size()) {
+    		curr.addAll(Arrays.asList(lore));
+    	} else {
+    		curr.addAll(index, Arrays.asList(lore));
+    	}
+    	setLore(curr);
+    }
+    
+    /**
+     * Appends lore to the item.
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: White space needed

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 5]
```diff
@@ -56,4 +56,30 @@
...
+    
+    /**
+     * Adds lore to the item starting from a specific index.
+     * @param index Index to begin at.
+     * @param lore Lore to be added.
+     * @see #addLore(String...)
+     */
+    default void addLore(int index, String... lore) {
+    	List<String> curr = lore();
+    	if (curr.isEmpty() || index == curr.size()) {
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: Reverse logic, fix index statement to match the javadoc please.

**Comment by AgentTroll** [comment]
Added line comments

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 6]
```diff
@@ -56,4 +56,30 @@
...
+    		curr.addAll(Arrays.asList(lore));
+    	} else {
+    		curr.addAll(index, Arrays.asList(lore));
+    	}
+    	setLore(curr);
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: You have forgotten to return, this results in two writes to setLore

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java (hunk 5 above)
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: You may also want to add that to the javadoc instead if it results in desired functionality

**Comment by AgentTroll** [comment]
Added line comments

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java (hunk 6 above)
Comment (nickrobson) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: wat. that's an if-else statement. there's only one setLore call

**Comment by nickrobson** [comment]
@AgentTroll This should be fixed.

**Comment by AgentTroll** [comment]
What do you mean with "this" should be fixed?

**Comment by nickrobson** [comment]
I mean the PR's content.

**Comment by nickrobson** [comment]
@AgentTroll pls

**Comment by AgentTroll** [comment]
What do you want me todo with it?

**Comment by nickrobson** [comment]
idk, magic?

**Comment by nickrobson** [comment]
I'm pretty sure I've fixed the things you asked me to fix.

**Comment by AgentTroll** [comment]
Invite me back to the Skype chat

**Comment by nickrobson** [comment]
troll.dude.3?

**Comment by AgentTroll** [comment]
Yes

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 7]
```diff
@@ -56,4 +59,32 @@
...
+     * Adds lore to the item starting from a specific index.    
+     * @param index Index to begin insertion at. If equal to number of lore, it appends.
+     * @param lore Lore to be added.
+     * @see #addLore(String...)
+     */
+    default void addLore(int index, String... lore) {
+        List<String> curr = lore();
+        if (curr == null)
+            curr = Lists.newArrayList();
+        if (curr.isEmpty() || index == curr.size()) {
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: The technical index check can be a hard detail to find. I'd say remove it, and change your implementation of `addLore(String[])`

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 8]
```diff
@@ -56,4 +59,32 @@
     default void setLore(String... lore) {
         setLore(Arrays.asList(lore));
     }
+    
+    /**
+     * Adds lore to the item starting from a specific index.
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: Add whitespace here

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java [hunk 9]
```diff
@@ -56,4 +59,32 @@
...
+        if (curr.isEmpty() || index == curr.size()) {
+            curr.addAll(Arrays.asList(lore));
+        } else {
+            curr.addAll(index, Arrays.asList(lore));
+        }
+        setLore(curr);
+    }
+    
+    /**
+     * Appends lore to the item.
```
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: Add whitespace here

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java (hunk 7 above)
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: TL;DR you don't need the second OR check, just remove that and stay with isEmpty.

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java (hunk 8 above)
Comment (nickrobson) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: There's no whitespace in that area in any other javadocs in the project.

**Comment by nickrobson** [comment]
I'm going to reopen this.

File: src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java (hunk 8 above)
Comment (AgentTroll) [inline src/main/java/net/tridentsdk/meta/item/ItemDisplayProperties.java]: There is.

**Pull request closed without merging**